Pocket/pocket-ios
# feat(account-deletion): adding in account deletion options

## Summary

Creates the basic scaffolding for:
* Account Management Screen
* Delete Account Screen
* Button Styles

## References 

- IN-773

## Implementation Details
* I took time to build out Pocket Button styles within the Textile class including pressed states which we did not have.
* I had to implement a custom toggle style for the checkboxes that is in Textile
* I also worked to ensure we could start to use swift previews by following the [opinion piece here](https://medium.com/swift2go/swiftui-opinion-viewmodel-doesnt-belong-in-previews-62d9e1485b38)


## Test Steps

- [ ] Given I am a user
 And I click settings
 I can see Account Management
 
- [ ] Given I am a user
 And I click settings
And I click Account Management
I can see Delete Account

- [ ] Given I am a user
 And I click settings
And I click Account Management
And I click on Delete Account
I can see a delete confirmation view

- [ ] Given I am a free user
 And navigate to delete confirmation view
I see a check box to confirm I know it will delete my account

- [ ] Given I am a premium user
 And navigate to delete the confirmation view
I see a check box to confirm I know it will delete my account
and I see a checkbox to confirm I have canceled premium

- [ ] Given I am a user
 And navigate to delete the confirmation view
And I have clicked the checkboxes
The delete account button becomes clickable

- [ ] Given I am a premium user
 And navigate to delete the confirmation view
And I have clicked the I have canceled premium checkbox
The deleting pocket help button disappears


- [ ] Given I am a premium user
 And navigate to delete the confirmation view
And I click the cancelling premium help button
I am taken to a help article within pocket on how to cancel premium

## PR Checklist:

## Screenshots



https://user-images.githubusercontent.com/1010384/222850497-9c60378f-bbd4-44fd-97f4-54ff00b63e04.mov



https://user-images.githubusercontent.com/1010384/222850530-6e920a15-ae70-488d-a60d-cc15fd595d5d.mov
